knative-operator/deploy/crds/operator_v1alpha1_knativekafka_crd.yaml

```diff
@@ -0,0 +1,120 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: knativekafkas.operator.serverless.openshift.io
+spec:
+  group: operator.serverless.openshift.io
+  names:
+    kind: KnativeKafka
+    listKind: KnativeKafkaList
+    plural: knativekafkas
+    singular: knativekafka
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            channel:
+              description: Allows configuration for KafkaChannel installation
+              properties:
+                bootstrapServers:
+                  description: BootstrapServers is comma separated string of bootstrapservers
+                    that the KafkaChannels will use
+                  type: string
+                enabled:
+                  description: Enabled defines if the KafkaChannel installation is
+                    enabled
+                  type: boolean
+                setAsDefaultChannelProvisioner:
```

Comment:
should we add some filed to define what the actual shape of a default channel is ? 

e.g. for:
```
  numPartitions: 3
  replicationFactor: 1
```

Which than would translate to this CM value on the server:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: default-ch-webhook
  namespace: knative-eventing
data:
  # Configuration for defaulting channels that do not specify CRD implementations.
  default-ch-config: |
    clusterDefault:
      apiVersion: messaging.knative.dev/v1beta1
      kind: KafkaChannel
      spec:
        numPartitions: 3
        replicationFactor: 1
```

Reply:
Ok, this comment of yours made me remember something: https://issues.redhat.com/browse/SRVKE-397 and https://issues.redhat.com/browse/SRVKE-487

We should actually not have this field anymore, remember the discussion @matzew?

Reply:
Our operator shouldn't go and touch that configmap because then we have an operator war over that configmap.

Let me know what you think and I will delete that afterwards.

Reply:
I thought we agreed on some shape like:

```
Spec:
  source: 
    enable: true
  channel:
    enable: true
    default: true
    bootstrapServers: foo.bar:9092, baz.bla:9092
```
since the `KnativeEventing` operator CR owns the `default-ch-webhook` CM, we should translate the values from the `KnativeKafka` spec to it. Or do we expect users to touch the "ConfigMap" syntax on the `KnativeEventing` CR, after they _enabled_ the channel in the `KnativeKafka` CR ?

Reply:
fwiw: `KnativeKafka` probably should not mutate the `KnativeEventing` CR since it has no ownership over it.

Reply:
yeah, we had the issue in the past - I thought it's easier on a combined Operator.

OK... let's not do the `default` here ...

Reply:
Well the `KnativeEventing` controller could read `KnativeKafka` and use it as "configuration" for its own reconcilation. That way, it'd totally work.

Reply:
OK, that would be good.

That way we set the "relevant" info on the `KnativeKafka` CR, and do _NOT_ ask the user to use some funky ConfigMap sytanx to set the previous installed channel as the default.  

OK, that would be better UX, I guess

Reply:
> yeah, we had the issue in the past - I thought it's easier on a combined Operator.

The upstream operator we use is the owner of that configmap, not our operator.

> Well the KnativeEventing controller could read KnativeKafka and use it as "configuration" for its own reconcilation. That way, it'd totally work.

I am not clear on this one. @markusthoemmes can you explain this a bit more please?

Downstream KnativeEventing controller doesn't own the configmap, so it also cannot change; same problem. 
It can change the KnativeEventing CR's config field (https://github.com/knative/operator/blob/master/docs/configuration.md#specconfig) so that the upstream operator can adjust the configmap. But, not sure if this is a good idea to change the CR except in a webhok.

Reply:
that's right, good point @aliok. The operator extension design will make this possible though. I'd punt on it for now.

Reply:
Ok, let's try to merge this PR for now and we can still think about it later.
I will remove that field now